### PR TITLE
Do not createOrder / configure checkout until BoltCheckout is defined

### PR DESCRIPTION
**Pull request by danac-gs**
# Description
Fix for the case when connect.js executes after plugin javascript.

Fixes: (link Asana Task)

(Optional: Add a changelog by writing a comment after "#changelog" on the next line)

\#changelog

# Type of change

- [x] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [x] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: view/frontend/templates/js/replacejs.phtml [hunk 1]
```diff
@@ -218,6 +218,33 @@ ob_start();
     }());
     ////////////////////////////////////////////////////////////////////////
 
+    ////////////////////////////////////////////////////////////////////////
+    // Wait for an object to be defined and
+    // execute a callback when it becomes available
+    ////////////////////////////////////////////////////////////////////////
+    ! function() {
```
Comment (daisy1754) [inline view/frontend/templates/js/replacejs.phtml]: no need for wrapping with `! function() { }()`? You would usually want to do this to narrow variable scope but below you are defining global function anyway

File: view/frontend/templates/js/replacejs.phtml [hunk 2]
```diff
@@ -218,6 +218,33 @@ ob_start();
...
+        window.whenDefined = function(obj, property, callback) {
+            var prop = '_'+property;
```
Comment (daisy1754) [inline view/frontend/templates/js/replacejs.phtml]: move this under else

File: view/frontend/templates/js/replacejs.phtml [hunk 3]
```diff
@@ -218,6 +218,33 @@ ob_start();
...
+            if (obj[property]) {
+                callback();
+            } else {
+                Object.defineProperty(obj, property, {
```
Comment (daisy1754) [inline view/frontend/templates/js/replacejs.phtml]: add comment (calls callback when variable is set)

File: view/frontend/templates/js/replacejs.phtml (hunk 1 above)
Comment (danac-gs) [reply view/frontend/templates/js/replacejs.phtml]: done

File: view/frontend/templates/js/replacejs.phtml (hunk 2 above)
Comment (danac-gs) [reply view/frontend/templates/js/replacejs.phtml]: dome

File: view/frontend/templates/js/replacejs.phtml (hunk 3 above)
Comment (danac-gs) [reply view/frontend/templates/js/replacejs.phtml]: done

**Comment by danac-gs** [comment]
> can you point master branch?

changed

**Comment by danac-gs** [comment]
@daisy1754 can you take a look again?

**Pull request merged**